kratos/sources/registry.cpp

```diff
@@ -0,0 +1,127 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Pooyan Dadvand
+//
+
+// System includes
+#include <mutex>
+
+// External includes
+
+// Project includes
+#include "includes/registry.h"
+
+namespace Kratos
+{
+
+namespace {
+    std::once_flag flag_once;
+}
+
+    RegistryItem* Registry::mspRootRegistryItem = nullptr;
+
+    RegistryItem& Registry::GetItem(std::string const& ItemFullName){
+        const std::lock_guard<LockObject> scope_lock(ParallelUtilities::GetGlobalLock());
+
+        auto item_path = StringUtilities::SplitStringByDelimiter(ItemFullName, '.');
+        KRATOS_ERROR_IF(item_path.empty()) << "The item full name is empty" << std::endl;
+
+        
+        RegistryItem* p_current_item = &GetRootRegistryItem();
+
+        for(std::size_t i = 0 ; i < item_path.size() ; i++){
+            auto& item_name = item_path[i];
+            if(p_current_item->HasItem(item_name)){
+                p_current_item = &p_current_item->GetItem(item_name);
+            }
+            else{
+                KRATOS_ERROR << "The item \"" << ItemFullName << "\" is not found in the registry. The item \"" << p_current_item->Name() << "\" does not have \"" << item_name << "\"" << std::endl;
+            }
+        }
+
+        return *p_current_item;
+    }
+
+    void Registry::RemoveItem(std::string const& ItemFullName){
+        const std::lock_guard<LockObject> scope_lock(ParallelUtilities::GetGlobalLock());
+
+        auto item_path = StringUtilities::SplitStringByDelimiter(ItemFullName, '.');
+        KRATOS_ERROR_IF(item_path.empty()) << "The item full name is empty" << std::endl;
+        
+        RegistryItem* p_current_item = &GetRootRegistryItem();
+
+        for(std::size_t i = 0 ; i < item_path.size() - 1 ; i++){
+            auto& item_name = item_path[i];
+            if(p_current_item->HasItem(item_name)){
+                p_current_item = &p_current_item->GetItem(item_name);
+            }
+            else{
+                KRATOS_ERROR << "The item \"" << ItemFullName << "\" is not found in the registry. The item \"" << p_current_item->Name() << "\" does not have \"" << item_name << "\"" << std::endl;
+            }
+        }
+
+        auto& item_name = item_path.back();
+        if(p_current_item->HasItem(item_name)){
+            p_current_item->RemoveItem(item_name);
+        }
```

Comment:
```suggestion
        } else {
```

Reply:
There is no instruction in the style guide about this. Leaving as it is to avoid polluting current production ready code.